controller/src/main/java/io/pravega/controller/store/stream/Batcher.java

```diff
@@ -0,0 +1,19 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.store.stream;
+
+/**
+ * Batcher interface to facilitate callers to group their data into different batches.
+ * Implementation of this interface should manage the batch identifier and update it's value
+ * based on their business logic.
+ */
+public interface Batcher<T> {
```

Comment:
This really isn't a very good abstraction. Perhaps some of the other methods could be moved here so this interface is more meaningful?

Reply:
This abstraction is separation of batching with garbage collection.

The store needs to know which batch to write the data to and batchers independently update the batches based on their business logic and store simply gets the latest batch by requesting over this method. 

It could have been a functional interface too but I added this extra class in case we want more ability in future.
I can make it a functional interface if you feel we dont think it is meaningful in its current avatar.